tmva/tmva/inc/TMVA/DNN/Architectures/Cpu/CpuTensor.h

```diff
@@ -196,7 +196,7 @@ class TCpuTensor : public TMVA::Experimental::RTensor<AFloat, TCpuBuffer<AFloat>
    //this will be an unsafe view. Method exists for backwards compatibility only
    TCpuMatrix<AFloat> GetMatrix() const
    {
-      size_t ndims = 0;
+      [[maybe_unused]] size_t ndims = 0;
```

Comment:
I think the warning is actually correct, running the loop be low is pointless if the assert is off.  I would consider using:
```
assert( [&shape]() { int ndims; for (auto& shape_i : shape) { if (shape_i != 1) { ndims++; } }; return ndims; } () <= 2 && shape.size() > 1); // to support shape cases {n,1}
```
(alternative use a `[[maybe_unused]]` variable for the lambda).